cmake/scripts/common/ArchSetup.cmake

```diff
@@ -157,3 +157,9 @@ if(CMAKE_BUILD_TYPE STREQUAL "Debug")
   add_options (ALL_LANGUAGES DEBUG "-g" "-D_DEBUG" "-Wall")
 endif()
 
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_options(CXX "DEBUG;RELEASE" "-Wno-overloaded-virtual")
+  if(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" AND (CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 7 OR CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 7))
```

Comment:
clang >= 3.5 is supposed to support this. I guess we should do this on all platforms, not just OSX?
Also those options should only be enabled due to -Wall in Debug builds, so disabling in Release shouldn't be necessary?